### DISCUSSION TITLE
初めて寄付が追加した後ユーザーページの生成されている文言をみて、どう行動していいかわからなくなってしまった

### DISCUSSION THREAD

**Discussion by yamanoku** [azu/philan.net#42] [Q&A] [answered]
寄付の記録づけとして、philan.net 使わせていただいているものです。

![Screen Shot 2021-05-01 at 11 38 59](https://user-images.githubusercontent.com/1996642/116768481-db2b3380-aa71-11eb-9b74-0baf9ac2f306.png)

寄付の追加後「記録に成功しました」という追加後ページが表示されたのですが、
下部に「ユーザーページを生成中です」という文言が表示されます。

> 最大1~2分程度かかります

というのとspinnerアイコンが表示されていることで、
最初寄付したときはこのページで待機していないといけないのか、と思って画面をそのままにしておりました。

追加後のページ自体は、遷移してすぐ離脱してしまっても特に問題ないのでしょうか。

**Comment by azu** [comment, marked as answer]
ページにはアクセスしても特に問題ないです。
実際にはキャッシュの設定が1分になっているので、すぐアクセスした場合には最新の内容ではなく一つ前のキャッシュが表示されるかもしれないという意味合いですね。
([Next.jsのISRで動的コンテンツをキャッシュするときの戦略](https://zenn.dev/catnose99/articles/8bed46fb271e44)みたいな投稿者本人はキャッシュなしで取得する仕組みを実装するのが一番いいですね)

**Reply by yamanoku** [reply]
ご回答ありがとうございます。キャッシュとのこと、了解いたしました 👍🏼 

> 実際にはキャッシュの設定が1分になっているので、すぐアクセスした場合には最新の内容ではなく一つ前のキャッシュが表示されるかもしれないという意味合いですね。

**Reply by azu** [reply]
#46 と https://github.com/azu/philan.net/pull/49 で、
自分自身のページの場合はキャッシュを無視して再取得する仕組みを入れて、Spinnerを表示する必要性がなくなったので、寄付後ページの表示を変更しました。

![image](https://user-images.githubusercontent.com/19714/125808615-f4452833-e72a-408e-a3a4-1d62f37c1e3c.png)
